src/ir/from_ast/build_ctx.rs

```diff
@@ -62,11 +62,15 @@ pub(super) struct BuildCtx<'prog> {
     sigs: &'prog SigMap,
 
     // Mapping from names to IR nodes.
-    event_map: ScopeMap<ir::Event>,
-    inst_map: ScopeMap<ir::Instance>,
-    inv_map: ScopeMap<ir::Invoke>,
-    port_map: ScopeMap<ir::Port, InvPort>,
-    param_map: ScopeMap<ir::Param>,
+    event_map: ScopeMap<ir::EventIdx>,
```

Comment:
Question about data structures similar to ScopeMap (where we want to track a mapping to and/or from IR structures): should they take something like `Event` (and internally use `Idx<Event>`, which is what `Foreign` does) as a type or `EventIdx` (as done here)? I've seen it both ways throughout the code, and it might be good to make a consistent decision on this?

Reply:
Yeah, this was how the interface originally worked by I ended up changing it to be generic scope tracking doesn't seem like it has anything to do with `Idx`. We should probably extract this into a utilities file somewhere instead of keeping it here since it can be generally useful.